graylog2-server/src/main/java/org/graylog/plugins/views/startpage/StartPageService.java

```diff
@@ -68,26 +68,32 @@ public PaginatedResponse<LastOpened> findLastOpenedFor(final SearchUser searchUs
                 .items()
                 .stream()
                 .skip((long) (page - 1) * perPage)
+                .map(i -> startPageItemTitleRetriever
+                        .retrieveTitle(i.grn())
+                        .map(title -> new LastOpened(i.grn(), title, i.timestamp()))
+                        .orElse(null))
+                .filter(Objects::nonNull)
                 .limit(perPage)
-                .map(i -> new LastOpened(i.grn(), catalog.getTitle(i.grn()), i.timestamp()))
                 .toList();
 
         return PaginatedResponse.create("lastOpened", new PaginatedList<>(items, items.size(), page, perPage));
     }
 
-    private String getTitle(RecentActivityDTO i) {
-        return i.itemTitle() == null ? catalog.getTitle(i.itemGrn()) : i.itemTitle();
-    }
-
     public PaginatedResponse<RecentActivity> findRecentActivityFor(final SearchUser searchUser, int page, int perPage) {
         final var items = recentActivityService.findRecentActivitiesFor(searchUser, page, perPage);
         final var mapped = items.stream()
-                 .map(i -> new RecentActivity(i.id(),
-                        i.activityType(),
-                        i.itemGrn(),
-                        getTitle(i),
-                        i.userName(),
-                        i.timestamp())).toList();
+                .map(i -> startPageItemTitleRetriever
+                        .retrieveTitle(i.itemGrn())
+                        .map(title -> new RecentActivity(i.id(),
+                                i.activityType(),
+                                i.itemGrn(),
+                                title,
+                                i.userName(),
+                                i.timestamp()))
+                        .orElse(null)
```

Comment:
Related to last comment:
```suggestion
                        .stream()
```